apps/server/src/modules/tldraw/domain/ws-shared-doc.do.ts

```diff
@@ -0,0 +1,65 @@
+import { Doc } from 'yjs';
+import WebSocket from 'ws';
+import { Awareness, encodeAwarenessUpdate } from 'y-protocols/awareness';
+import { encoding } from 'lib0';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { WSMessageType } from '@src/modules/tldraw/types/connection-enum';
+import { TldrawWsService } from '@src/modules/tldraw/service';
+
+// disable gc when using snapshots!
+const gcEnabled: boolean = Configuration.get('TLDRAW__GC_ENABLED') as boolean;
+
+export class WsSharedDocDo extends Doc {
+	name: string;
+
+	public conns: Map<WebSocket, Set<number>>;
+
+	public awareness: Awareness;
+
+	/**
+	 * @param {string} name
+	 * @param {TldrawWsService} tldrawService
+	 */
+	constructor(name: string, private tldrawService: TldrawWsService) {
+		super({ gc: gcEnabled });
+		this.name = name;
+		this.conns = new Map();
+		this.awareness = new Awareness(this);
+		this.awareness.setLocalState(null);
+
+		this.awareness.on('update', this.awarenessChangeHandler);
+		this.on('update', (update: Uint8Array, origin, doc: WsSharedDocDo) => {
+			this.tldrawService.updateHandler(update, origin, doc);
+		});
+	}
+
+	/**
+	 * @param {{ added: Array<number>, updated: Array<number>, removed: Array<number> }} changes
+	 * @param {WebSocket | null} conn Origin is the connection that made the change
+	 */
+	public awarenessChangeHandler = (
+		{ added, updated, removed }: { added: Array<number>; updated: Array<number>; removed: Array<number> },
+		conn: WebSocket | null
+	) => {
+		const changedClients = added.concat(updated, removed);
+		if (conn !== null) {
+			const connControlledIDs = this.conns.get(conn) as Set<number>;
+			if (connControlledIDs !== undefined) {
+				added.forEach((clientID) => {
+					connControlledIDs.add(clientID);
+				});
+				removed.forEach((clientID) => {
+					connControlledIDs.delete(clientID);
+				});
+			}
+		}
+		// broadcast awareness update
+		const encoder = encoding.createEncoder();
```

Comment:
This method include 3 orchstration parts if i see it correctly. 
1. something with connections
2. parts of coding
3. send stuff to connections

Can you split this 3 parts into private methods with names that explain what they do and use them in this public method? Then we have seperate the logic from orchestration and improve that the code is more self explained.
Then the orec